flashlight/app/imgclass/examples/ImageNetTransformer.cpp

```diff
@@ -253,25 +268,71 @@ int main(int argc, char** argv) {
   int batchIdx = 0;
   for (int epoch = (FLAGS_exp_checkpoint_epoch + 1); epoch < FLAGS_train_epochs;
        epoch++) {
-    trainDataset.resample();
+    trainDataset->resample(epoch);
+    trainDataset1->resample(epoch + 4399);
 
     // Get an iterator over the data
     timeMeter.resume();
     int idx = 0;
-    for (auto& example : trainDataset) {
+    // for (auto& example : trainDataset) {
+    for (; idx < trainDataset->size();) {
+      auto example = trainDataset->get(idx);
+      auto example1 = trainDataset1->get(idx);
       lrScheduler(epoch, batchIdx++);
       opt.zeroGrad();
+
       // Make a Variable from the input array.
-      auto inputs = noGrad(example[kImagenetInputIdx]);
+      Variable inputs;
+      float mixP =
+          static_cast<float>(std::rand()) / static_cast<float>(RAND_MAX);
+      float lambda =
+          static_cast<float>(std::rand()) / static_cast<float>(RAND_MAX);
+      float example1Weight = FLAGS_train_p_mixup;
```

Comment:
should be sampled from the beta(train_p_mixup, train_p_mixup), so that mixing happens with random coeff, see https://github.com/facebookresearch/mixup-cifar10/blob/master/train.py#L119.

Also you can mixup inside batch only as in original implementation. But I like mixing across the whole dataset.